src/main/java/uk/gov/justice/digital/delius/service/UserAccessService.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.justice.digital.delius.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.digital.delius.data.api.AccessLimitation;
+import uk.gov.justice.digital.delius.helpers.CurrentUserSupplier;
+
+import java.util.Collection;
+import java.util.Set;
+
+@Service
+@AllArgsConstructor
+public class UserAccessService {
+    private final UserService userService;
+    private final OffenderService offenderService;
+    private final CurrentUserSupplier currentUserSupplier;
+    @Value("${user-access.scopes.dont-apply-exclusions-for}")
+    private final Set<String> ignoreExclusionRoles;
+    @Value("${user-access.scopes.dont-apply-restrictions-for}")
+    private final Set<String> ignoreRestrictionRoles;
+
+    public void checkExclusionsAndRestrictions(String crn, Collection<? extends GrantedAuthority> authorities) {
+        final var username = currentUserSupplier.username();
+        final var offender = offenderService.getOffenderByCrn(crn);
```

Comment:
I think you could move this down after we have established we need to do the check, else  I think you will be making the additional call when the scope says you don't need to check

Reply:
I originally did have it like that but then having changed to the scoped configuration I figured our default case is that we will need to check so moved this up so it's only done once. I take your point that in that case it will be redundant though so will update.

Reply:
One thing to remember is the biggest client of this API is the elastic search indexer that definitely won't have any restrictions. So that call would be redundant in the vast majority of cases.
One mitigation is the call *might* be cached by Hibernate - but not too sure.

Reply:
My reading on it suggests that it does but I couldn't swear to it. From what I understand it will cache for the DB session and I'd have thought that should be request scoped? It's all happening synchronously so seems unlikely that it wouldn't reuse the session anyway...